scripts/spack_packages/umpire/package.py

```diff
@@ -118,290 +83,149 @@ class Umpire(CMakePackage, CudaPackage):
+    @property
+    def cache_name(self):
+        hostname = socket.gethostname()
+        if "SYS_TYPE" in env:
+            hostname = hostname.rstrip('1234567890')
+        return "{0}-{1}-{2}@{3}.cmake".format(
+            hostname,
+            self._get_sys_type(self.spec),
+            self.spec.compiler.name,
+            self.spec.compiler.version
+        )
+
+    def spec_uses_toolchain(self, spec):
+        gcc_toolchain_regex = re.compile(".*gcc-toolchain.*")
+        using_toolchain = list(filter(gcc_toolchain_regex.match, spec.compiler_flags['cxxflags']))
+
+        return using_toolchain
+
+    def spec_uses_gccname(self, spec):
+        gcc_name_regex = re.compile(".*gcc-name.*")
+        using_gcc_name = list(filter(gcc_name_regex.match, spec.compiler_flags['cxxflags']))
 
+        return using_gcc_name
+
+
+
+    def initconfig_compiler_entries(self):
+        spec = self.spec
+        entries = super(Umpire, self).initconfig_compiler_entries()
+
+        entries.append(cmake_cache_option("ENABLE_FORTRAN", 
+            ('+fortran' in spec) and (self.compiler.fc is not None)))
+        entries.append(cmake_cache_option("ENABLE_C", '+c' in spec))
+        
         fortran_compilers = ["gfortran", "xlf"]
-        if any(compiler in f_compiler for compiler in fortran_compilers) and ("clang" in cpp_compiler):
-            cfg.write(cmake_cache_entry("BLT_CMAKE_IMPLICIT_LINK_DIRECTORIES_EXCLUDE",
+        if any(compiler in self.compiler.fc for compiler in fortran_compilers) and ("clang" in self.compiler.cxx):
+            entries.append(cmake_cache_string("BLT_CMAKE_IMPLICIT_LINK_DIRECTORIES_EXCLUDE",
             "/usr/tce/packages/gcc/gcc-4.9.3/lib64;/usr/tce/packages/gcc/gcc-4.9.3/gnu/lib64/gcc/powerpc64le-unknown-linux-gnu/4.9.3;/usr/tce/packages/gcc/gcc-4.9.3/gnu/lib64;/usr/tce/packages/gcc/gcc-4.9.3/lib64/gcc/x86_64-unknown-linux-gnu/4.9.3"))
 
             libdir = pjoin(os.path.dirname(
-                           os.path.dirname(f_compiler)), "lib")
+                           os.path.dirname(self.compiler.fc)), "lib")
             flags = ""
             for _libpath in [libdir, libdir + "64"]:
                 if os.path.exists(_libpath):
                     flags += " -Wl,-rpath,{0}".format(_libpath)
             description = ("Adds a missing libstdc++ rpath")
             if flags:
-                cfg.write(cmake_cache_entry("BLT_EXE_LINKER_FLAGS", flags,
-                                            description))
+                entries.append(cmake_cache_string("BLT_EXE_LINKER_FLAGS", flags, description))
 
 
-        gcc_toolchain_regex = re.compile(".*gcc-toolchain.*")
-        gcc_name_regex = re.compile(".*gcc-name.*")
-
-        using_toolchain = list(filter(gcc_toolchain_regex.match, spec.compiler_flags['cxxflags']))
-        using_gcc_name = list(filter(gcc_name_regex.match, spec.compiler_flags['cxxflags']))
         compilers_using_toolchain = ["pgi", "xl", "icpc"]
```

Comment:
I see icpc, so I wonder: pgc++ ? xlc++ ?

Reply:
I think the prefixes will catch that when I try and do the matching.